apps/client/src/common/utils/aliases.ts

```diff
@@ -26,4 +29,42 @@ export const validateAlias = (alias: string) => {
   }
 
   return valid;
-};
\ No newline at end of file
+};
+
+export const getAliasRoute = (location: Location, data: Alias[], searchParams: URLSearchParams) => {
+  const currentURL = location.pathname.substring(1);
+  let redirectURL = '';
+  // we need to check if the whole url here is an alias, so we can redirect
+  const foundAlias = data.filter((d) => d.alias === currentURL && d.enabled)[0];
+  if (foundAlias) {
+    redirectURL = generateURLFromAlias(foundAlias);
+    return redirectURL;
+  } else {
+    for (const d of data) {
+      const aliasOnPage = searchParams.get('alias');
+      if (aliasOnPage) {
+        // if the alias fits the alias on this page, but the URL is diferent, we redirect user to the new URL
+        // if we have the same alias and its enabled and its not empty
+        if (d.alias !== '' && d.enabled && d.alias === aliasOnPage) {
+          const newAliasPath = resolvePath(d.pathAndParams);
+          const urlParams = new URLSearchParams(newAliasPath.search);
+          urlParams.set('alias', d.alias);
```

Comment:
I think we might be able to pass the whole thing to the constructor and let it deduplicate?
```
    const urlParams = new URLSearchParams({
      ...newAliasPath.search,
      alias: aliasData.alias,
    });
```

Reply:
`newAliasPath.search` is a string, so it cant be spread.